com.unity.multiplayer.mlapi/Runtime/Core/SnapshotRTT.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using UnityEngine;
+
+namespace MLAPI
+{
+    internal class ClientRtt
+    {
+        /// <summary>
+        /// Round-trip-time data
+        /// </summary>
+        public struct Rtt
+        {
+            public double Best;
+            public double Average;
+            public double Worst;
+            public int SampleCount;
+        }
+        public ClientRtt()
+        {
+            m_RttSendTimes = new double[k_RingSize];
+            m_SendKey = new int[k_RingSize];
+            m_MeasuredLatencies = new double[k_RingSize];
+        }
+
+        /// <summary>
+        /// Returns the Round-trip-time computation for this client
+        /// </summary>
+        public Rtt GetRtt()
+        {
+            var ret = new Rtt(); // is this a memory alloc ? How do I get a stack alloc ?
+            var index = m_LatenciesBegin;
+            double total = 0.0;
+            ret.Best = m_MeasuredLatencies[m_LatenciesBegin];
+            ret.Worst = m_MeasuredLatencies[m_LatenciesBegin];
+
+            while (index != m_LatenciesEnd)
+            {
+                total += m_MeasuredLatencies[index];
+                ret.SampleCount++;
+                ret.Best = Math.Min(ret.Best, m_MeasuredLatencies[index]);
+                ret.Worst = Math.Max(ret.Worst, m_MeasuredLatencies[index]);
+                index = (index + 1) % k_RttSize;
+            }
+
+            if (ret.SampleCount != 0)
+            {
+                ret.Average = total / ret.SampleCount;
+            }
+            else
+            {
+                ret.Average = 0;
+                ret.Best = 0;
+                ret.Worst = 0;
+                ret.SampleCount = 0;
+            }
+
+            return ret;
+        }
+
+        internal void NotifySend(int key, double when)
+        {
+            m_RttSendTimes[key % k_RingSize] = when;
+            m_SendKey[key % k_RingSize] = key;
+        }
+
+        internal void NotifyAck(int key, double when)
```

Comment:
does `key` mean `sequence` or would it wrap at some number and start from 0 again?
also, `when` vs `whenSec` vs `timeSec`?
I think little details/optics like these matter when someone else wants to read/debug your code :)